core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnBatch.scala

```diff
@@ -146,18 +146,17 @@ final class ColumnBatchIterator(region: LocalRegion, val batch: ColumnBatch,
   protected var currentVal: ByteBuffer = _
   private var currentDeltaStats: ByteBuffer = _
   private var currentKeyPartitionId: Int = _
-  private var currentKeyUUID: String = _
+  private var currentKeyUUID: Long = _
```

Comment:
Can do but does not matter. This will not be read unless assigned by iterator next, and if not present then hasNext has to be false which callers should check first.